### PR TITLE
Viewer for powsybl-network-area-diagram

### LINKED ISSUE #32: Create viewer for powsybl-network-area-diagram
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
No viewer for the images generated by [powsybl-network-area-diagram](https://github.com/powsybl/powsybl-network-area-diagram/)

**What is the expected behavior?**
Similarly to the single-line-diagram-viewer of powsybl-dev-tools, having a debug viewer to test powsybl-network-area-diagram

**What is the motivation / use case for changing the behavior?**
Helping develop and test powsybl-network-area-diagram

**Other information** 
- Most of the code might be taken from single-line-diagram-viewer in this same repo, as this viewer is expected to be very similar. The single-line-diagram-viewer has known many changes and therefore some of its code might now be useless/irrelevant, so don't hesitate to change it if looks odd to you.
- The user can select either:
    - the full network to be displayed by ticking "full network" checkbox: then the SVG will be generated using `new NetworkAreaDiagram(network).draw(stringWriter)`
    - only part of the network by unticking that box, selecting a voltage level and choosing a depth: then the SVG will be generated using `new NetworkAreaDiagram(network, voltageLevelId, depth).draw(stringWriter)`
- Note that when full network checkbox is ticked, the part below can be hidden.
- When a node (voltage level) is clicked, information about that voltage level should be displayed in the infos part. A button near that infos part should be added to generate a diagram centered on the clicked node.
- A button "Run loadflow" should be added, as often the power flow values in the input file are missing. The corresponding code to call is `LoadFlow.run(network);` and the following artifactId of load flow implementation should be added in the `pom.xml`: `powsybl-open-loadflow`
- LayoutParameters and SvgParameters should be modifiable, similarly to single-line-diagram-viewer
- Layout, LabelProvider and StyleProvider can be changed with drop-down list, but those lists will contain only one element (the three of them have only one implementation so far)
- Following input files can be used to test:
     * [Sample files](https://github.com/powsybl/powsybl-cgmes-conformity-assessments/raw/main/src/test/resources/TestConfigurations_packageCASv2.0.zip) from the ENTSO-E
     * IEEE [test case files](https://github.com/powsybl/powsybl-core/tree/main/psse/psse-model/src/test/resources) 

Here's a screen mock-up of what's expected.
mockup-nad-viewer

### PR DESCRIPTION
**Pull request by louislhotte**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines

**Does this PR already have an issue describing the problem ?**
Fixes #32

### CODE REVIEW

File: network-area-diagram-viewer/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-parent</artifactId>
+        <version>4</version>
+        <relativePath/>
+    </parent>
+
+    <groupId>org.example</groupId>
```
Comment (flo-dup) [inline network-area-diagram-viewer/pom.xml:14]: you should remove this

File: network-area-diagram-viewer/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    <artifactId>powsybl-network-area-diagram-viewer</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (flo-dup) [inline network-area-diagram-viewer/pom.xml:16]: Even if we do not plan to release this, let's start with a 0.1.0-SNAPSHOT version

File: network-area-diagram-viewer/pom.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        <java.version>11</java.version>
+        <franzxaver.version>0.1</franzxaver.version>
+        <javafx.version>17</javafx.version>
+        <slf4j.version>1.7.22</slf4j.version>
+        <junit.version>4.13.1</junit.version>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <powsybl.nad.version>0.4.0-SNAPSHOT</powsybl.nad.version>
+        <powsybl.core.version>4.8.0-SNAPSHOT</powsybl.core.version>
+        <powsybl.open.loadflow.version>0.18.0-SNAPSHOT</powsybl.open.loadflow.version>
```
Comment (flo-dup) [inline network-area-diagram-viewer/pom.xml:29]: We want to test the latest changes of network-area-diagram, but not really the latest from powsybl-core and powsybl-open-loadflow. Hence, keep the snapshot version only for network-area-diagram

File: network-area-diagram-viewer/pom.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-open-loadflow</artifactId>
+            <version>${powsybl.open.loadflow.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.openjfx</groupId>
+            <artifactId>javafx-controls</artifactId>
+            <version>17.0.1</version>
```
Comment (flo-dup) [inline network-area-diagram-viewer/pom.xml:81]: add this version number in the properties

File: network-area-diagram-viewer/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Area-diagram-viewer-rte
```
Comment (flo-dup) [inline network-area-diagram-viewer/README.md:1]: you can remove RTE name

File: network-area-diagram-viewer/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+Most updated version of the area diagram viewer
+
+The area diagram viewer uses these three repositories which versions should be changed (in the pom.xml file) if different from the ones below:
+```
+powsybl.nad.version: 0.4.0-SNAPSHOT
+powsybl.core.version: 4.8.0-SNAPSHOT
+powsybl.open.loadflow.version: 0.18.0-SNAPSHOT
```
Comment (flo-dup) [inline network-area-diagram-viewer/README.md:9]: You should update this following previous comment

File: network-area-diagram-viewer/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Area-diagram-viewer-rte
+
+Most updated version of the area diagram viewer
```
Comment (flo-dup) [inline network-area-diagram-viewer/README.md:3]: I think this readme file can stay there, but you should also add a subsection in the root README.md of powsybl-dev-tools.

**Review by flo-dup — CHANGES_REQUESTED** [review]
Just starting with some comments about the pom.xml and readme

**Review by flo-dup — CHANGES_REQUESTED** [review]
Each java file should start with 
```/**
 * Copyright (c) 2022, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
```
and also have a
```
 /**
 * @author Louis Lhotte <louis.lhotte@student-cs.fr>
 */
```
just before the definition of the class or interface

The pom.xml file should also include the license header (at line 2):
```
<!--

    Copyright (c) 2022, RTE (http://www.rte-france.com)
    This Source Code Form is subject to the terms of the Mozilla Public
    License, v. 2.0. If a copy of the MPL was not distributed with this
    file, You can obtain one at http://mozilla.org/MPL/2.0/.

-->
```

File: network-area-diagram-viewer/pom.xml (hunk 1 above)
Comment (louislhotte) [reply network-area-diagram-viewer/pom.xml:14]: you mean `<relativePath/>` and `<groupId>org.example</groupId>`?

File: network-area-diagram-viewer/src/main/java/com/powsybl/ad/viewer/controller/ControllerDiagram.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import javafx.stage.Stage;
+
+import java.io.StringWriter;
+
+/**
+ * @author Louis Lhotte <louis.lhotte@student-cs.fr>
+ */
+public class ControllerDiagram
+{
+    private Stage primaryStage;
```
Comment (flo-dup) [inline network-area-diagram-viewer/src/main/java/com/powsybl/ad/viewer/controller/ControllerDiagram.java:21]: You're not using a uniform style for your code, in fact there's a some rules in powsybl for that, which are described in a checkstyle config file: https://github.com/powsybl/powsybl-parent/tree/main/powsybl-build-tools/src/main/resources/powsybl-build-tools. I'll add the corresponding mvn plugin to this repo to check it automatically. Let's keep that esthetics issue for the end.

File: network-area-diagram-viewer/src/main/java/com/powsybl/ad/viewer/controller/ControllerImport.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            fileChooser.setTitle("Open case File");
+            File file = fileChooser.showOpenDialog(primaryStage);
+            if (file != null)
+            {
+                loadNetwork(file.toPath());
+                ControllerParameters.reselectDefaultChoiceBoxes();  // select first element of parameters when we import something new
+                try {
+                    cleanSVG();  // clean the window and the variables
+                    drawNetwork();  // changes the variable svgWriter
+                    DisplaySVG.loadContent(svgWriter.toString());  // changes the variable contentSVG
```
[the remark this answers is not in the document]
Comment (louislhotte) [reply network-area-diagram-viewer/src/main/java/com/powsybl/ad/viewer/controller/ControllerImport.java:69]: I was jsut loading contentSVG in memory but that was just not ideal. Everything's been refactored.

**Comment by louislhotte** [comment]
The pom.xml should not be merged into main.

**Comment by flo-dup** [comment]
> The pom.xml should not be merged into main (except for the ucte support that was added by flo-dup).

What do you mean? I suppose you mean ***from*** main into this PR, as you reverted the changes of the merge commit?

But why shouldn't we applied the changes from main? the goal of this tool is for developers to test the latest changes of the powsybl-network-area-diagram, hence the latest snapshot version: 0.5.0-SNAPSHOT.

**Comment by louislhotte** [comment]
> > The pom.xml should not be merged into main (except for the ucte support that was added by flo-dup).
> 
> What do you mean? I suppose you mean _**from**_ main into this PR, as you reverted the changes of the merge commit?
> 
> But why shouldn't we applied the changes from main? the goal of this tool is for developers to test the latest changes of the powsybl-network-area-diagram, hence the latest snapshot version: 0.5.0-SNAPSHOT.

Sorry, I didn't formulate it well. There is still one bugfix I would like to address so I will keep working on the oldest version until everything works perfectly and only then will I transition to the lastest version. Therefore, merging into main should not be done yet as the pom.xml is still the old one (that I still commited in case someone wants to test where the viewer is at).

**Comment by flo-dup** [comment]
Too many changes needed to clean the code, replaced by #54

**Pull request closed without merging**